test/transforms/cat_to_num_transform.py

```diff
@@ -123,3 +123,17 @@ def test_ordered_target_statistics_encoder(task_type):
         out.feat_dict[stype.numerical][:, :total_numerical_cols]).all())
     assert (dataset.tensor_frame.col_names_dict[stype.numerical] ==
             out.col_names_dict[stype.numerical][:total_numerical_cols])
+
+
+def test_cat_to_num_transform_with_target_contains_nans():
+    num_rows = 10
+    dataset: Dataset = FakeDataset(num_rows=num_rows, with_nan=False,
+                                   stypes=[stype.numerical, stype.categorical],
+                                   task_type=TaskType.REGRESSION,
+                                   create_split=True)
+    dataset.df.at[0, 'target'] = np.nan
```

Comment:
Why don't we modify `FakeDataset` so that `with_nan=True` gives nan values in the target as well>

Reply:
Makes sense. I hope making this change doesn't affect other test cases. 🥹